two-factor.php

```diff
@@ -24,6 +24,43 @@
 require_once( TWO_FACTOR_DIR . 'class.two-factor-core.php' );
 Two_Factor_Core::add_hooks();
 
+if ( version_compare( PHP_VERSION, '5.3.0', '<' ) ) {
+	/**
+	 * Remove FIDO U2F if PHP 5.2.
+	 *
+	 * @param array $providers Array of providers.
+	 * @return array Array of providers.
+	 */
+	function remove_fido_u2f_support( $providers ) {
+		unset( $providers['Two_Factor_FIDO_U2F'] );
+		return $providers;
+	}
+
+	add_filter( 'two_factor_providers', 'remove_fido_u2f_support' );
+
+	/**
+	 * Display PHP Upgrade Notice for FIDO U2F.
+	 */
+	function upgrade_php_nag() {
+		$screen = get_current_screen();
+		if ( in_array( $screen->id, array( 'profile', 'user-edit' ) ) ) {
+			?>
+			<div class="update-nag">
+				<?php
+					printf(
+						esc_html__( 'You are using too old version of PHP to use FIDO U2F as a Two-Factor Option. Please consider %1$supgrading PHP%2$s.' ),
+						'<a href="' . esc_url( 'http://php.net/supported-versions.php' ) . '">',
+						'</a>'
+					);
+				?>
+			</div>
+			<?php
+		}
+	}
+
+	add_action( 'admin_notices', 'upgrade_php_nag' );
```

Comment:
I'm not sure if we should add a nag, or just silently not include it.  I'm thinking perhaps not include the nag, and just silently drop the feature because Core won't keep it if it gets merged in.